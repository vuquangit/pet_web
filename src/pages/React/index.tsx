import React from 'react'

export function ReactPage() {
  return <div>ReactPage </div>
}
