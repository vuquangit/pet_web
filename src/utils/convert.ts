import humps from 'humps'
import { cloneDeep } from 'lodash'

export const convertObjectToFormData = (obj: any) => {
  // convert to FormData
  const formData = new FormData()
  const dataDecamelize = cloneDeep(humps.decamelizeKeys(obj))

  for (const key in dataDecamelize) {
    formData.append(key, dataDecamelize[key])
  }

  return formData
}
