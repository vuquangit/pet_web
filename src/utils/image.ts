const dataURItoBlob = (dataURI: string) => {
  const bytes =
    dataURI.split(',')[0].indexOf('base64') >= 0
      ? atob(dataURI.split(',')[1])
      : unescape(dataURI.split(',')[1])
  const mime = dataURI.split(',')[0].split(':')[1].split(';')[0]
  const max = bytes.length
  const ia = new Uint8Array(max)
  for (let i = 0; i < max; i += 1) ia[i] = bytes.charCodeAt(i)
  return new Blob([ia], { type: mime })
}

const blobToFile = (theBlob: Blob, fileName: string): File => {
  return new File(
    [theBlob as any], // cast as any
    fileName,
    {
      lastModified: new Date().getTime(),
      type: theBlob.type,
    }
  )
}

export const resizeImage = ({ file, maxSize }: { file: File; maxSize: number }) => {
  const reader = new FileReader()
  const image = new Image()
  const canvas = document.createElement('canvas')

  const resize = () => {
    let { width, height } = image

    if (width > height) {
      if (width > maxSize) {
        height *= maxSize / width
        width = maxSize
      }
    } else if (height > maxSize) {
      width *= maxSize / height
      height = maxSize
    }

    canvas.width = width
    canvas.height = height
    canvas.getContext('2d')?.drawImage(image, 0, 0, width, height)

    const dataUrl = canvas.toDataURL(file.type)

    const blob = dataURItoBlob(dataUrl)
    return blobToFile(blob, file.name)
  }

  return new Promise((resolve, reject) => {
    if (!file.type.match(/image.*/)) {
      reject(new Error('Not an image'))
      return
    }

    reader.onload = (readerEvent) => {
      image.onload = () => resolve(resize())
      // @ts-ignore
      image.src = readerEvent.target?.result
    }

    reader.readAsDataURL(file)
  })
}
