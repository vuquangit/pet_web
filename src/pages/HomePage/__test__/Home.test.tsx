import React from 'react'
import { HomePage } from '../index'
import { render } from '@/tests/unit/utils'
import { describe, expect, it } from '@jest/globals'

describe('HomePage tests', () => {
  it('renders the component', () => {
    const { asFragment, container } = render(<HomePage />)

    const loadingClass = container.getElementsByClassName('App')
    expect(loadingClass.length).toBe(1)

    expect(asFragment()).toMatchSnapshot()
  })
})
