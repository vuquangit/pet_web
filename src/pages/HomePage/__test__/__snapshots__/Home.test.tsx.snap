// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests renders the component 1`] = `
<DocumentFragment>
  <div
    class="App"
    data-cy="home-page"
  >
    <div
      class="App-header"
    >
      <svg
        class="App-logo mb-10 fill-gray-800 dark:fill-white"
        data-file-name="SvgLogo"
      />
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to reload.
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
      <button
        class="btn-default"
      >
        Test Unauthenticated
      </button>
    </div>
  </div>
</DocumentFragment>
`;
